File: NeetCode/65/65.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:

        hashAlph = [0] * 26

        for char in tasks:
            hashAlph[ord(char) - 64] += 1

        hashAlph.sort()
        maxf = hashAlph[25]
        idle = (maxf - 1) * n

        for i in range(24, -1, -1):
            idle -= min(maxf - 1, hashAlph[i])
        return max(0, idle) + len(tasks)

File: NeetCode/65/65.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        vector<int> hashAlph(26,0);

        for(auto task : tasks)
        {
            hashAlph.at(task - 'A')++;
        }

        sort(hashAlph.begin(), hashAlph.end());
        int maxFrequency = hashAlph.at(25);
        int idle = (maxFrequency - 1)*n;
        for(int i = 24; i >= 0; i--){
            idle -= min(maxFrequency - 1, hashAlph.at(i));
        }
        return max(0, idle) + tasks.size();
    }
};
